lib/utils.go

```diff
@@ -14,3 +14,13 @@ func ParseJSONArray(items string) ([]string, error) {
 
 	return parsedItems, nil
 }
+
+func runAsync(f func() error) <-chan error {
```

Comment:
I would keep it in `api` package. It can be imported into `lib` and we avoid duplication.